0021-zarf-values/README.md

```diff
@@ -0,0 +1,291 @@
+      onDeploy:
+        before:
+          - cmd: "echo '{ \"memory\": \"256Mi\", \"cpu\": \"200m\" }'"
+            setValues:
+              - path: component.resources.limits
+                type: json
+                
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will initially use the values provided for any components before the one with `setValues`
+**And Then** apply the values provided in the `setValues` action for any following components based on the actions lifecycle.
+
+> [!NOTE]
+> *This would apply to all `zarf-config` formats not just YAML
+
+### Risks and Mitigations
+
+This will introduce a wholly new way to input values into Zarf that will live alongside the existing Variables, Constants and Templates for now.  Because of this, the feature will need to be clearly disambiguated from Variables/Constants/Templates in documentation and while this feature should not introduce many breaking changes being implemented alongside the existing featureset, the feature to map Zarf Variables to Helm Values should be deprecated and removed in favor of the new Zarf Values mapping to assist with disambiguation.  If the feature gains traction and is accepted by the community, a deprecation plan for the original Zarf Variables/Constants/Templates should be created.
+
+This feature also could open up Zarf packages to being less declarative - especially if a package author opens up security-critical Helm values in their charts.  This caveat should be clearly documented as a concern which should also recommend a policy engine be used to enforce security-critical values within the cluster itself.
+
+Because we will be using more `interface{}` types, we should also look into the security implications of this feature and ensure that this is well tested and we utilize some of Helm's existing protections against `nil` maps and other potential security issues with this feature.
+
+## Design Details
+
+The new `values.files` field would be added to the `ZarfPackageConfig` schema and would accept a list of local values files or URLs, matching the existing functionality of the `valuesFiles` key under `charts`.  This key would also follow the same composability logic as the `valuesFiles` key with additional parent values files merging with and overriding any common keys from children imports.  Since this is a global field, values would be merged regardless of component similar to how variables work today.  The `values.schema` field would simply replace the child version if it were non-empty in the parent, similar to the `namespace` or `releaseName` fields in `charts` today.
```

Comment:
Could you make a user story that includes the `values.files` field set in the package definition?

Reply:
done